quantum/painter/qp_draw_codec.c

```diff
@@ -64,6 +65,21 @@ bool qp_internal_decode_recolor(painter_device_t device, uint32_t pixel_count, u
     return qp_internal_decode_palette(device, pixel_count, bits_per_pixel, input_callback, input_arg, qp_internal_global_pixel_lookup_table, output_callback, output_arg);
 }
 
+bool qp_internal_send_bytes(painter_device_t device, uint32_t byte_count, qp_internal_byte_input_callback input_callback, void* input_arg, qp_internal_byte_output_callback output_callback, void* output_arg) {
+    uint32_t remaining_bytes = byte_count;
+    while (remaining_bytes > 0) {
+        uint8_t byteval = input_callback(input_arg);
```

Comment:
This needs to be `int16_t` to keep the error information:
```suggestion
        int16_t byteval = input_callback(input_arg);
```
(And `qp_internal_decode_palette()` apparently has the same bug.)

Reply:
Good catch, I've fixed both here and for qp_internal_decode_palette.

I was also tempted to remove the `if (byteval < 0) { return false; }` lines from both functions as well because an unsigned integer can't be negative (so the complier should strip it out anyway), but I wasn't sure if perhaps there is a plan to change the return types to signed integers for error handling someday.